packages/vc-status-list/src/types/index.ts

```diff
@@ -23,33 +23,58 @@ import {
 } from '@veramo/core'
 import { DataSource } from 'typeorm'
 
-export interface CreateNewStatusListFuncArgs extends Omit<StatusList2021ToVerifiableCredentialArgs, 'encodedList'> {
-  correlationId: string
-  length?: number
+export type StatusList2021Args = {
+  encodedList: string
+  indexingDirection: StatusListIndexingDirection
+  statusPurpose?: StatusPurpose2021
+  // todo: validFrom and validUntil
 }
 
-export interface UpdateStatusListFromEncodedListArgs extends StatusList2021ToVerifiableCredentialArgs {
-  statusListIndex: number | string
-  value: boolean
+export type OAuth2StatusListArgs = {
+  status: 'active' | 'suspended' | 'revoked'
+  expiresAt?: string
 }
 
-export interface UpdateStatusListFromStatusListCredentialArgs {
-  statusListCredential: OriginalVerifiableCredential
+export type BaseCreateNewStatusListArgs = {
+  type: StatusListType
+  id: string
+  issuer: string | IIssuer
+  correlationId?: string
+  length?: number
+  proofFormat?: ProofFormat
   keyRef?: string
-  statusListIndex: number | string
-  value: boolean
+  statusList2021Args?: StatusList2021Args
+  oauth2StatusListArgs?: OAuth2StatusListArgs
 }
 
-export interface StatusList2021ToVerifiableCredentialArgs {
+export type UpdateStatusList2021Args = {
   issuer: string | IIssuer
   id: string
-  type?: StatusListType
   statusPurpose: StatusPurpose2021
   encodedList: string
+}
+
+export type UpdateOAuth2StatusListArgs = {
+  status: 'active' | 'suspended' | 'revoked'
```

Comment:
Please keep it symetrical with StatusList2021. So statusPurpose and extract the values. The fact you now already have the same values multiple times should have been an indicator

Reply:
fixed